File: projects/10/java/Penguin.java
/* Penguin.java provides a Penguin subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Ian Christensen, for CS 214 at Calvin College
 * Date: Spring, 2018
 */

public class Penguin extends Walking {

	public Penguin() {
		super();
	}
    
	/* explicit constructor
	 * Receive: name, a String 
	 * PostCond: myName == name.
	 */
	public Penguin(String name) {     
		super(name);
	} 

	/* A Penguin's Call
	 * Return: a Penguin-call ("Huh-huh-huh-huuuuh").
	 */
	public String call() {
		return "Huh-huh-huh-huuuuh";
	} 
}

File: projects/10/ruby/Penguin.rb
#!/usr/bin/ruby
# Penguin.rb Defines the Penguin class which inherits attributes and methods from the Bird superclass.
# Begun by: Dr. Adams, for CS 214 at Calvin College
# Completed by: Ian Christensen, for CS 214 at Calvin College
# Date: Spring, 2018

require './Walking.rb'

class Penguin < Walking

	# Method: call, accessor for a Duck call
	# Precondition: Duck must exist
	# Receives: a Duck object
	# Returns: 'Quack!', a duck call
	def call
		'Huh-huh-huh-huuuuh'
	end
end
